### PR TITLE
RenderPass based SSAO

### PR DESCRIPTION
**Pull request by mvaligursky**
- implementation of SSAO using render passes
- few small improvements compared to the implementation using the post-effects (increasing number of samples does not modify the brightness, exposed additional controls ..)

no SSAO
Screenshot 2024-06-04 at 16 28 04

un-blured SSAO
Screenshot 2024-06-04 at 16 28 14

blurred SSAO
Screenshot 2024-06-04 at 16 28 23

### CODE REVIEW

**Comment by Maksims** [comment]
`minAngle` does not seem to do anything?

File: src/extras/render-passes/render-pass-depth-aware-blur.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+ *
+ * @category Graphics
+ * @ignore
+ */
+class RenderPassDepthAwareBlur extends RenderPassShaderQuad {
+    constructor(device, sourceTexture) {
+        super(device);
+        this.sourceTexture = sourceTexture;
+
+        this.shader = this.createQuadShader('DepthAwareBlurShader', shaderChunks.screenDepthPS + /* glsl */`
```
Comment (slimbuck) [inline src/extras/render-passes/render-pass-depth-aware-blur.js:15]: these should probably be overridable chunks?

File: src/extras/render-passes/render-pass-ssao.js [hunk 2]
```diff
@@ -0,0 +1,362 @@
+import { ADDRESS_CLAMP_TO_EDGE, FILTER_NEAREST, PIXELFORMAT_R8 } from '../../platform/graphics/constants.js';
+import { RenderTarget } from '../../platform/graphics/render-target.js';
+import { Texture } from '../../platform/graphics/texture.js';
+import { RenderPassShaderQuad } from '../../scene/graphics/render-pass-shader-quad.js';
+import { shaderChunks } from '../../scene/shader-lib/chunks/chunks.js';
+import { RenderPassDepthAwareBlur } from './render-pass-depth-aware-blur.js';
+
+const fs = /* glsl */`
```
Comment (slimbuck) [inline src/extras/render-passes/render-pass-ssao.js:8]: same here

File: src/extras/render-passes/render-pass-depth-aware-blur.js (hunk 1 above)
Comment (mvaligursky) [reply src/extras/render-passes/render-pass-depth-aware-blur.js:15]: Not really. Making those chunks would stop their three-shaking. The render pass is a combination of shader and it's parameters, bot nothing else in the file.

If the users wants to use for example different depth aware blur, they can simply create a different render pass (shader + it's set up) and use that instead.

**Comment by mvaligursky** [comment]
> `minAngle` does not seem to do anything?

hey, last minute change for it to work as a preset. Fixed now, thanks @Maksims

**Review by slimbuck — APPROVED** [review]

**Pull request merged**